src/include/storage/storage_structure/db_file_utils.h

```diff
@@ -33,6 +34,11 @@ class DBFileUtils {
 public:
     constexpr static common::page_idx_t NULL_PAGE_IDX = common::INVALID_PAGE_IDX;
 
+    static uint8_t* pinPage(BMFileHandle& fileHandle, common::page_idx_t pageIdx,
```

Comment:
I'm a little concerned that this adds new ways of doing pinning/optimistic reads without making it obvious what the differences are at a glance.
Maybe `BufferManager::pin`/`BufferManager::optimisticRead` should be private, or maybe we shouldn't even be passing around the BufferManager at all for these file operations. The BMFileHandle already stores a pointer to the BufferManager, so we could have a `BMFileHandle::pin(page_idx_t, PageReadPolicy)` function instead that handles the memory mode. While that doesn't really make the differences any clearer, it at least would make the BMFileHandle functions the more obvious choice, and we could remove the BufferManager arguments from DBFileUtils and avoid passing it around as much and make the BufferManager more of an internal detail that you shouldn't need to interact with when working with file handles (though #3743 will still mean that the MemoryManager needs to get passed around, but with this removed maybe we can even remove `MemoryManager::getBufferManager`).